libs/native/src/main/java/org/elasticsearch/nativeaccess/WindowsFunctions.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess;
+
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.nativeaccess.lib.Kernel32Library;
+
+/**
+ * Native functions specific to the Windows operating system.
+ */
+public class WindowsFunctions {
+    private static final Logger logger = LogManager.getLogger(Systemd.class);
+
+    private final Kernel32Library kernel;
+
+    WindowsFunctions(Kernel32Library kernel) {
+        this.kernel = kernel;
+    }
+
+    /**
+     * Retrieves the short path form of the specified path.
+     *
+     * @param path the path
+     * @return the short path name, or the original path name if unsupported or unavailable
+     */
+    public String getShortPathName(String path) {
+        String longPath = "\\\\?\\" + path;
+        // first we get the length of the buffer needed
+        final int length = kernel.GetShortPathNameW(longPath, null, 0);
+        if (length == 0) {
+            logger.warn("failed to get short path name: {}", kernel.GetLastError());
+            return path;
+        }
+        final char[] shortPath = new char[length];
+        // knowing the length of the buffer, now we get the short name
+        if (kernel.GetShortPathNameW(longPath, shortPath, length) > 0) {
+            assert shortPath[length - 1] == '\0';
+            return new String(shortPath, 0, length - 1);
+        } else {
+            logger.warn("failed to get short path name: {}", kernel.GetLastError());
+            return path;
+        }
+    }
+
+    /**
+     * Adds a Console Ctrl Handler for Windows. On non-windows this is a noop.
+     *
+     * @return true if the handler is correctly set
+     */
+    public boolean addConsoleCtrlHandler(ConsoleCtrlHandler handler) {
+        return kernel.SetConsoleCtrlHandler(dwCtrlType -> {
```

Comment:
The original JNA code had a list of callbacks, with a comment 
``` 
// Callbacks must be kept around in order to be able to be called later,
// when the Windows ConsoleCtrlHandler sends an event.
```
I see you are omitting it; I suppose the idea was to avoid getting them garbage collected, but probably a callback is already referenced by the code instantiating them (in this case Elasticsearch.java)? It's hard to tell (at least, for me) what happens in the managed/un-managed transition. Who is keeping a reference tho the callback/handler/lambda?

Reply:
At least on the FFI side, I don't think this is necessary, albeit with a slight change. I had previously assumed the JDK would retain a link through the upcall stub, but I realized that that depends on the lifetime of the arena where the native address is allocated. Yet the current branch is using an auto arena, so I think it may be GC'd. I've changed this to the global arena, so it should retain the handler for the lifetime of the jvm.

On the JNA side, I guess this is necessary, I don't see how it would ensure the handler isn't GC'd, so I'll add it back inside there.

Reply:
> I've changed this to the global arena, so it should retain the handler for the lifetime of the jvm.

This makes sense, the JVM/GC has to have a way of knowing things can or cannot be collected